src/components/Datatable/Datatable.tsx

```diff
@@ -134,17 +140,28 @@ const Datatable = <D extends Record<string, unknown>>({
     [onDataFetch, appliedFilters, searchQuery],
   );
 
+  const handleOnSuggestionsFetch = useCallback(
+    async (query: string) => {
+      setFilterSuggestions(await onSuggestionsFetch(query));
+    },
+    [onSuggestionsFetch],
+  );
+
   const handleOnSearch = useCallback(
-    (query: string) => {
-      setSearchQuery(query);
-      onDataFetch({
-        pageSize: defaultPageSize,
-        pageIndex: 0,
-        query,
-        filters: appliedFilters,
-      });
+    async (queryValue: string) => {
+      setSearchQuery(queryValue);
+      if (controlsConfig.searchConfig?.hasSuggestions === true) {
+        handleOnSuggestionsFetch(queryValue);
+      } else {
+        onDataFetch({
+          pageSize: defaultPageSize,
+          pageIndex: 0,
+          query: searchQuery,
```

Comment:
it can't use the `searchQuery`  it leads to that it uses old state, there needs to be used the arg which is passed into this function, this was already fixed once